x-pack/filebeat/input/cometd/input.go

```diff
@@ -0,0 +1,445 @@
+	workerOnce   sync.Once          // Guarantees that the worker goroutine is only started once.
+	workerWg     sync.WaitGroup     // Waits on worker goroutine.
+
+	ackedCount *atomic.Uint32                   // Total number of successfully ACKed messages.
+	Transport  httpcommon.HTTPTransportSettings `config:",inline"`
+	Retry      retryConfig                      `config:"retry"`
+	out        chan TriggerEvent
+}
+
+// TriggerEvent describes an event received from Bayeaux Endpoint
+type TriggerEvent struct {
+	Channel  string `json:"channel"`
+	ClientID string `json:"clientId"`
+	Data     struct {
+		Event struct {
+			CreatedDate time.Time `json:"createdDate"`
+			ReplayID    int       `json:"replayId"`
+			Type        string    `json:"type"`
+		} `json:"event"`
+		Object json.RawMessage `json:"payload"`
+	} `json:"data,omitempty"`
+	Successful bool `json:"successful,omitempty"`
+}
+
+// Status is the state of success and subscribed channels
+type Status struct {
+	channels  []string
+	clientID  string
+	connected bool
+}
+
+type BayeuxHandshake []struct {
+	ClientID string `json:"clientId"`
+	Channel  string `json:"channel"`
+	Ext      struct {
+		Replay bool `json:"replay"`
+	} `json:"ext"`
+	MinimumVersion           string   `json:"minimumVersion"`
+	Successful               bool     `json:"successful"`
+	SupportedConnectionTypes []string `json:"supportedConnectionTypes"`
+	Version                  string   `json:"version"`
+}
+
+type Subscription struct {
+	ClientID     string `json:"clientId"`
+	Channel      string `json:"channel"`
+	Subscription string `json:"subscription"`
+	Successful   bool   `json:"successful"`
+}
+
+type Credentials struct {
+	AccessToken string `json:"access_token"`
+	InstanceURL string `json:"instance_url"`
+	IssuedAt    string `json:"issued_at"`
+	ID          string `json:"id"`
+	TokenType   string `json:"token_type"`
+	Signature   string `json:"signature"`
+}
+
+type clientIDAndCookies struct {
+	clientID string
+	cookies  []*http.Cookie
+}
+
+// Bayeux struct allow for centralized storage of creds, ids, and cookies
+type Bayeux struct {
+	creds Credentials
+	id    clientIDAndCookies
+}
+
+type retryConfig struct {
+	MaxAttempts *int           `config:"max_attempts"`
+	WaitMin     *time.Duration `config:"wait_min"`
+	WaitMax     *time.Duration `config:"wait_max"`
+}
+
+type Event struct {
+	EventId string `json:"EventIdentifier"`
+}
+
+func (c Credentials) bayeuxUrl() string {
+	return c.InstanceURL + "/cometd/" + cometdVersion
+}
+
+// Call is the base function for making bayeux requests
+func (b *Bayeux) call(body string, route string) (resp *http.Response, e error) {
+	var jsonStr = []byte(body)
+	req, err := http.NewRequest("POST", route, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return nil, fmt.Errorf("bad Call request: %v", err)
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", b.creds.AccessToken))

```

Comment:
Depending on the implementation, Salesforce oriented or general, this can be optional, right?

Reply:
We are not quite sure about this one.